assets/js/theme/custom/mega-menu.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Open mega menu on search button click
+ */
+
+import { constants } from './constants'
+
+let isShowMoreEventAttached = false
+
+export default function megaMenu(hasAttachedEvent) {
+    let hasEvent = hasAttachedEvent
+    const { MOBILE_WIDTH } = constants
+    const dropdown = document.querySelector('.js-dropdown')
+    const searchButton = document.querySelector('.js-mega-menu-search')
+    const searchInput = document.querySelector('.js-search-quick')
+    const showMoreTriggers = document.querySelectorAll('.js-show-more-item')
+
+    const openDesktopMenu = () => {
+        dropdown.classList.toggle('is-open')
+        searchInput.focus()
+    }
+
+    const updateEventListeners = () => {
+        const windowWidth = window.innerWidth
+        if (searchButton) {
+            if (windowWidth >= MOBILE_WIDTH && !hasEvent) {
+                searchButton.addEventListener('click', openDesktopMenu)
+                hasEvent = true
+            }
+        }
+    }
+
+    const showMoreMenuItems = () => {
+        showMoreTriggers.forEach((item) => {
+            const button = item.querySelector('.js-show-more')
+            if (button && !isShowMoreEventAttached) {
```

Comment:
if the `isShowMoreEventAttached` is `true` I think we could skip iterating through buttons at all?